### PR TITLE
Distributed feature

### PR DESCRIPTION
**Pull request by rameshka**
## Automation tests
 >

## Security checks
 >

### CODE REVIEW

File: components/org.wso2.carbon.das.jobmanager.core/src/main/java/org/wso2/carbon/das/jobmanager/core/topology/SiddhiTopologyCreatorImpl.java [hunk 1]
```diff
@@ -364,124 +359,178 @@ private void checkQueryType(InputStream inputStream, boolean queryElement) {
...
-                                                         siddhiQueryGroupsList.get(j).getParallelism()));
+                                                Math.max(siddhiQueryGroup1.getParallelism(),
+                                                         siddhiQueryGroup2.getParallelism()));
 
-                                        siddhiQueryGroupsList.get(j).getInputStreams().get(key)
+                                        siddhiQueryGroup2.getInputStreams().get(key)
                                                 .getSubscriptionStrategy().setOfferedParallelism(
-                                                Math.max(siddhiQueryGroupsList.get(i).getParallelism(),
-                                                         siddhiQueryGroupsList.get(j).getParallelism()));
+                                                Math.max(siddhiQueryGroup1.getParallelism(),
```
Comment (tishan89) [inline components/org.wso2.carbon.das.jobmanager.core/src/main/java/org/wso2/carbon/das/jobmanager/core/topology/SiddhiTopologyCreatorImpl.java]: Are we handling the scenario of having n partition subscribers with different parallelism.

File: components/org.wso2.carbon.das.jobmanager.core/src/test/java/org/wso2/carbon/das/jobmanager/core/SiddhiTopologyCreatorTestCase.java [hunk 2]
```diff
@@ -27,6 +28,8 @@
 import java.util.List;
 
 public class SiddhiTopologyCreatorTestCase {
+    private static final Logger log = Logger.getLogger(SiddhiTopologyCreator.class);
```
Comment (tishan89) [inline components/org.wso2.carbon.das.jobmanager.core/src/test/java/org/wso2/carbon/das/jobmanager/core/SiddhiTopologyCreatorTestCase.java]: Get the correct logger

File: pom.xml [hunk 3]
```diff
@@ -1273,6 +1273,37 @@
                 <version>${apache.commons.text.version}</version>
             </dependency>
 
+            <!--Extension dependencies-->
```
Comment (tishan89) [inline pom.xml]: I already added these. Please get a pull

File: components/org.wso2.carbon.das.jobmanager.core/src/main/java/org/wso2/carbon/das/jobmanager/core/topology/SiddhiTopologyCreatorImpl.java (hunk 1 above)
Comment (rameshka) [reply components/org.wso2.carbon.das.jobmanager.core/src/main/java/org/wso2/carbon/das/jobmanager/core/topology/SiddhiTopologyCreatorImpl.java]: yes..n partition subscribers with different parallelism for the same key

**Pull request merged**